frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.tsx

```diff
@@ -96,7 +102,11 @@ const TaskComponent: React.FC<TaskProps> = (props) => {
       >
         <div className="odc-pipeline-vis-task__content">
           <div className={cx('odc-pipeline-vis-task__title', { 'is-text-center': !status })}>
-            {task.metadata ? task.metadata.name : ''}
+            {props.name
+              ? props.name
+              : task.metadata && task.metadata.name
+              ? task.metadata.name
+              : task.metadata.name}
```

Comment:
@abhinandan13jan I don't see why both true and false condition has the same variable `task.metadata.name`. Rest all looks good to me

Reply:
updated